### ISSUE TITLE
java.lang.AssertionError: User ids MUST be self-certified at this point!!

### ISSUE THREAD

**Issue by dschuermann** [open-keychain/open-keychain#1261]
From GPlay

"Crash during some kind of re-import?"
"Shuts down while consolidating every time it starts since the last update."
"new version crashes on start, trying to re-import keys."
```
java.lang.AssertionError: User ids MUST be self-certified at this point!!
at org.sufficientlysecure.keychain.provider.ProviderHelper.saveCanonicalizedPublicKeyRing(ProviderHelper.java:653)
at org.sufficientlysecure.keychain.provider.ProviderHelper.savePublicKeyRing(ProviderHelper.java:929)
at org.sufficientlysecure.keychain.operations.ImportExportOperation.importKeyRings(ImportExportOperation.java:299)
at org.sufficientlysecure.keychain.provider.ProviderHelper.consolidateDatabaseStep2(ProviderHelper.java:1280)
at org.sufficientlysecure.keychain.provider.ProviderHelper.consolidateDatabaseStep2(ProviderHelper.java:1205)
at org.sufficientlysecure.keychain.service.KeychainIntentService.onHandleIntent(KeychainIntentService.java:271)
at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:65)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:136)
at android.os.HandlerThread.run(HandlerThread.java:61)
```

**Comment by dschuermann** [comment]
Key with only v3 self-sigs: https://hkps.pool.sks-keyservers.net/pks/lookup?op=vindex&search=0xBF31CACA&fingerprint=on

**Comment by Valodim** [comment]
I think I got it. Taking a break for now but will look at it later

**Comment by Valodim** [comment]
got it reproduced in a key, and fixed in b2f32827c66c5cd5d14452b6f909cfe8c7c595a2. provided this is the only cause, at least.
